### PR TITLE
Implement customization of `rolePrefix` in `LdapUserDetailsManager`

### LINKED ISSUE #2083: SEC-1877: Allow injection of rolePrefix to LdapUserDetailsManager
[Yale Madden](https://jira.spring.io/secure/ViewProfile.jspa?name=ymadden@fruit.com) (Migrated from [SEC-1877](https://jira.spring.io/browse/SEC-1877?redirect=false)) said:

The rolePrefix member of LdapUserDetailsManager defaults to "ROLE_." This forces me to use this default rolePrefix throughout my application. Otherwise, convertAuthorityToGroup does not remove my rolePrefix and calls to createUser and updateUser return the following error:

javax.naming.NameNotFoundException: [LDAP: error code 32 - No Such Object]; remaining name 'cn=myprefix_user,ou=groups'
    at com.sun.jndi.ldap.LdapCtx.mapErrorCode(Unknown Source)
    at com.sun.jndi.ldap.LdapCtx.processReturnCode(Unknown Source)
    at com.sun.jndi.ldap.LdapCtx.processReturnCode(Unknown Source)
    at com.sun.jndi.ldap.LdapCtx.c_modifyAttributes(Unknown Source)
    at com.sun.jndi.toolkit.ctx.ComponentDirContext.p_modifyAttributes(Unknown Source)
    at com.sun.jndi.toolkit.ctx.PartialCompositeDirContext.modifyAttributes(Unknown Source)
    at javax.naming.directory.InitialDirContext.modifyAttributes(Unknown Source)
    at org.springframework.security.ldap.userdetails.LdapUserDetailsManager$5.executeWithContext(LdapUserDetailsManager.java:331)
    at org.springframework.ldap.core.LdapTemplate.executeWithContext(LdapTemplate.java:807)
    at org.springframework.ldap.core.LdapTemplate.executeReadWrite(LdapTemplate.java:802)
    at org.springframework.security.ldap.userdetails.LdapUserDetailsManager.modifyAuthorities(LdapUserDetailsManager.java:323)
    at org.springframework.security.ldap.userdetails.LdapUserDetailsManager.addAuthorities(LdapUserDetailsManager.java:315)
    at org.springframework.security.ldap.userdetails.LdapUserDetailsManager.updateUser(LdapUserDetailsManager.java:274)

Please provide a public setter for the rolePrefix member.

### PR DESCRIPTION
**Pull request by Haarolean**
Resolves #2083

### CODE REVIEW

File: ldap/src/main/java/org/springframework/security/ldap/userdetails/LdapUserDetailsManager.java [hunk 1]
```diff
@@ -403,6 +403,15 @@ public void setSecurityContextHolderStrategy(SecurityContextHolderStrategy secur
 		this.securityContextHolderStrategy = securityContextHolderStrategy;
 	}
 
+	/**
+	 * Sets the role prefix used when converting authorities. The default value is "ROLE_"
+	 * @param rolePrefix role prefix
+	 * @since 6.3
+	 */
+	public void setRolePrefix(String rolePrefix) {
```
Comment (jzheaux) [inline ldap/src/main/java/org/springframework/security/ldap/userdetails/LdapUserDetailsManager.java:411]: Will you please check for null in this method? You can use `Assert#notNull` for this.

**Review by jzheaux — CHANGES_REQUESTED** [review]
Thanks, @Haarolean! In addition to the inline feedback, will you please add some tests to `LdapUserDetailsManagerTests` for the new feature? And, will you please squash and format any commits into a single commit like so:

```
Commit Title (for example, Add LdapUserDetailsManager#setRolePrefix)

Closes gh-2083
```

**Comment by jzheaux** [comment]
Hi, @Haarolean, are you able to make the requested changes? I'm happy to help, if not.

**Comment by Haarolean** [comment]
> Hi, @Haarolean, are you able to make the requested changes? I'm happy to help, if not.

Hey, sure am! Been busy a bit lately, will push the changes today.

File: ldap/src/main/java/org/springframework/security/ldap/userdetails/LdapUserDetailsManager.java (hunk 1 above)
Comment (Haarolean) [reply ldap/src/main/java/org/springframework/security/ldap/userdetails/LdapUserDetailsManager.java:411]: done

**Comment by Haarolean** [comment]
@jzheaux do we have any instructions on how to enable checkstyle properly? Having a hard time with checkstyle (as always) in spring projects, I've imported the `etc/checkstyle/checkstyle.xml` as a scheme into IDEA via chsckstyle plugin, but despite that some things like import ordering are not being fixed automatically.

**Review by jzheaux — APPROVED** [review]

**Comment by jzheaux** [comment]
Thanks again, @Haarolean! This is now merged into `main`, including one polish commit to restore the Acegi header (we only change the ones in the form `20xx-20yy` for now).

**Comment by Haarolean** [comment]
@jzheaux thank you! Didn't know about the header part.

**Pull request merged**